apps/real-time-collaboration/server/AppInstanceLambda/index.js

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT-0
+
+'use strict';
+const AWS = require('aws-sdk');
+var response = require('cfn-response');
+AWS.config.update({region: process.env.AWS_REGION});
+const chime = new AWS.Chime({region: process.env.AWS_REGION});
+
+exports.handler = async (event, context) => {
+  console.log('Event: \n', event);
+  console.log('Create Chime SDK App Instance');
+
+  try {
+    if (event["RequestType"] !== "Create") {
+      await response.send(event, context, response.SUCCESS, {});
+    }
+
+    const dateNow = new Date();
+    let chime_response;
+    //create a chime app instance
+    const params = {
+      Name: 'AWSChimeMessagingSDKDemo-' + dateNow.getHours().toString() + dateNow.getMinutes().toString(),
+    };
+    chime_response = await chime.createAppInstance(
+      params,
+      function (err, data) {
+        if (err) console.log(err, err.stack);
+        // an error occurred
+        else {
+          console.log(data); // successful response
+          return data;
+        }
+      }
+    ).promise();
+
+    //Create AppInstanceAdmin
+    const createUserParams = {
+      AppInstanceArn: chime_response.AppInstanceArn, /* required */
+      AppInstanceUserId: 'ServiceUser', /* required */
+      ClientRequestToken: dateNow.getHours().toString() + dateNow.getMinutes().toString(), /* required */
+      Name: 'ServiceUser', /* required */
+    };
+    await chime.createAppInstanceUser(
+      createUserParams,
+      function (err, data) {
+        if (err) console.log(err, err.stack); // an error occurred
+        else {
+          console.log(data);           // successful response
+          chime_response.AdminUserArn = data.AppInstanceUserArn;
```

Comment:
For example, one downside with using a promise and a callback together is that it's unclear when the callback is executed. (A callback depends on the API author's implementation.)

```
// e.g., Does this line get executed first and then proceed to the next line? Or proceed to the next line first and then invoke the callback asynchronously? 
chime_response.AdminUserArn = data.AppInstanceUserArn;
```